forem/forem
# RFC 50: Remove hardcoded paths and use path_helpers instead (in the specs)

This was a painfully laborious PR 😭 

## What type of PR is this? (check all applicable)


☝️ It's a combination of all of these, I guess 🤔

## Description

These are a series of PR's and so it's difficult to provide context but I'm going to try and summarise it: 

As part of restructuring our admin navigation, we've changed our routes around. Soon we will be removing the old routes (and the feature flag) in favour of the new routes that contain scopes. These can be found in https://github.com/forem/forem/blob/main/config/routes/admin.rb.  You can find more info about the RFC [here](https://github.com/forem/rfcs/pull/50)

Hence, I've done the following in this PR:
- I've removed all the hardcoded routes and replaced them with the rails path helper (which also points to the new routes once the old routes are removed)
- I've updated the RSpec description, and context blocks to contain the new routes. Right now its a bit of a mismatch because with the feature flag disabled the tests are pointing to the old routes, but [without making these descriptions dynamic](https://forem-team.slack.com/archives/C7GE84DEJ/p1619467183497000) as well, this is the best we can do. In any case we planning on wrapping this feature up this week so it won't be too long until the path helpers are also referencing the new routes. Hence, I think that this is acceptable. 

What is not covered in this PR?
- I haven't yet updated the cypress routes to point to the new routes since we can't use path helpers in this part of the code base, hence I'll change it when I delete all the old routes.

## Related Tickets & Documents
https://github.com/forem/rfcs/pull/50

## QA Instructions, Screenshots, Recordings
- Double check that mappings are correct from hardcoded routes to dynamic path helpers 
- Optional: Grep through the repository for any hardcoded /admin routes left behind that need to be changed to path helpers. These routes will not include the user related, permissions and invitation routes as they have not changed so I've not updated all of them to use path helpers as yet. You can reference [this file](https://github.com/forem/forem/blob/main/config/routes/admin.rb) for routes that would have needed to change. This will also exclude the nested sidebar. 

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [x] No, and this is why: I changed to path helpers so need to ensure that all tests path.

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/oEnTTI3ZdK6ic/giphy.gif)
